src/main/java/info/matai/services/server/ServerVersioningService.java

```diff
@@ -1,16 +1,128 @@
 package info.matai.services.server;
 
-import info.matai.models.enums.ServerTypes;
+import info.matai.App;
+import info.matai.models.configuration.Server;
+import info.matai.services.server.instances.PaperFetcherService;
+import org.springframework.util.StringUtils;
 
 public class ServerVersioningService {
-  public static String mcVersion;
-  public static ServerTypes serverType;
+  public static Server plugins;
+  public static Server pluginsLock;
 
-  public final String SERVER_JARS_API_URL = "";
+  public ServerVersioningService() {
+    this.initData();
+    IServerFetcherService fetcher = getFetcherService();
+    if (plugins != null) {
+      initWithPluginJson(fetcher);
+    } else if (pluginsLock != null) {
+      this.initWithPluginLockJson(fetcher);
```

Comment:
Vu que l'autre méthode est définis au même niveau, je crois que tu pourrais juste `initWithPluginLockJson(fetcher);` sans le `this` pour être uniforme